java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/TextDocumentServiceImpl.java

```diff
@@ -2402,6 +2411,69 @@ protected CallHierarchyOutgoingCall createResultItem(CallHierarchyItem item, Lis
         return t.processRequest();
     }
 
+    @Override
+    public CompletableFuture<List<InlayHint>> inlayHint(InlayHintParams params) {
+        String uri = params.getTextDocument().getUri();
+        JavaSource js = getJavaSource(uri);
+
+        if (js != null) {
+            ConfigurationItem conf = new ConfigurationItem();
+            conf.setScopeUri(uri);
+            conf.setSection(NETBEANS_INLAY_HINT);
+            return client.configuration(new ConfigurationParams(Collections.singletonList(conf))).thenApply(c -> {
+                Set<String> enabled = new HashSet<>();
+                if (c != null && !c.isEmpty()) {
+                    JsonArray actualSettings = ((JsonArray) c.get(0));
+                    for (JsonElement el : actualSettings) {
+                        enabled.add(((JsonPrimitive) el).getAsString());
+                    }
+                } else {
+                    enabled.addAll(Arrays.asList("chained", "parameter", "var"));
+                }
+                List<InlayHint> result = new ArrayList<>();
+                try {
+                    js.runUserActionTask(cc -> {
+                        cc.toPhase(JavaSource.Phase.RESOLVED);
+                        Preferences preferences = NbPreferences.root().node("/org/netbeans/modules/java/editor/InlineHints/default");
+                        preferences.putBoolean(JAVA_INLINE_HINT_PARAMETER_NAME, enabled.contains("parameter"));
```

Comment:
`Preferences.put`!? Quite a hacky solution. I'd replace this with "branding API"_ and specified different default in `nbcode` suite.